Tests/FluentAssertions.Equivalency.Specs/DataRowSpecs.cs

```diff
@@ -1,332 +1,426 @@
 
-                // Act
-                Action action = () => dataRow1.Should().BeEquivalentTo(dataRow2);
+            var dataRow1 = dataSet1.TypedDataTable1[0];
+            var dataRow2 = dataSet2.TypedDataTable1[0];
 
-                // Assert
-                action.Should().Throw<XunitException>().Which.Message.Should().Contain("HasErrors");
-            }
+            dataRow2.String = Guid.NewGuid().ToString();
+            dataSet2.AcceptChanges();
 
-            [Fact]
-            public void When_HasErrors_does_not_match_and_property_is_excluded_it_should_succeed()
-            {
-                // Arrange
-                var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+            // Act
+            Action action = () => dataRow1.Should().BeEquivalentTo(dataRow2);
+
+            // Assert
+            action.Should().Throw<XunitException>().Which.Message.Should().Contain(dataRow2.String);
+        }
 
-                var dataSet2 = new TypedDataSetSubclass(dataSet1);
+        [Fact]
+        public void When_data_row_data_does_not_match_but_the_column_is_excluded_then_equivalency_test_should_succeed()
+        {
+            // Arrange
+            var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
 
-                var dataRow1 = dataSet1.TypedDataTable1[0];
-                var dataRow2 = dataSet2.TypedDataTable1[0];
+            var dataSet2 = new TypedDataSetSubclass(dataSet1);
 
-                dataRow2.RowError = "Manually added error";
+            var dataRow1 = dataSet1.TypedDataTable1[0];
+            var dataRow2 = dataSet2.TypedDataTable1[0];
 
-                // Act & Assert
-                dataRow1.Should().BeEquivalentTo(dataRow2, config => config.Excluding(dataRow => dataRow.HasErrors));
-            }
+            dataRow2.DateTime = DateTime.UtcNow;
+            dataSet2.AcceptChanges();
 
-            [Fact]
-            public void When_RowState_does_not_match_and_property_is_not_excluded_it_should_fail()
-            {
-                // Arrange
-                var dataSet1 = CreateDummyDataSet<TypedDataSetSubclass>();
+            // Act & Assert
+            dataRow1.Should().BeEquivalentTo(dataRow2, config => config.ExcludingColumn(dataSet2.TypedDataTable1.DateTimeColumn));
+        }
 
-                var dataSet2 = new TypedDataSetSubclass(dataSet1);
+        [Fact]
+        public void When_data_row_has_column_then_asserting_that_it_has_that_column_should_succeed()
+        {
+            // Arrange
+            var dataSet = CreateDummyDataSet<TypedDataSetSubclass>();
 
-                var dataRow1 = dataSet1.TypedDataTable1[0];
-                var dataRow2 = dataSet2.TypedDataTable1[0];
+            var dataRow = dataSet.TypedDataTable1[0];
 
-                dataRow1.DateTime = DateTime.UtcNow;
-                dataRow2.DateTime = dataRow1.DateTime;
+            string expectedColumnName = dataSet.TypedDataTable1.Columns.OfType<DataColumn>().Last().ColumnName;
```

Comment:
Could this be `.Cast<DataColumn>()` instead?
That avoids unexpectedly leaving any `nulls` out.

I see that `DataRowEquivalencyStep` and `DataSetEquivalencyStep` uses `OfType` as wel..

Reply:
I don't think it's possible to put a `null` into any of the System.Data collections. `OfType` and `Cast` should be functionally equivalent in this context. I can change them to `Cast`.